src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```diff
@@ -117,12 +109,14 @@ public void run() {
           if (responseCode >= 400) {
             LOG.warn("Http Error Code: {} for {}, \nReason: {}", responseCode, mUrl.toString(),
                     response.getStatusLine().getReasonPhrase());
-            mAccount.authenticate();
-            request.removeHeaders("X-Auth-Token");
-            request.addHeader("X-Auth-Token", mAccount.getAuthToken());
-            LOG.warn("Token recreated for {}.  Retry request", mUrl.toString());
-            response = client.execute(request);
-            responseCode = response.getStatusLine().getStatusCode();
+            if (responseCode == 401) { // Unauthorized error
+              mAccount.authenticate();
+              request.removeHeaders("X-Auth-Token");
+              request.addHeader("X-Auth-Token", mAccount.getAuthToken());
+              LOG.warn("Token recreated for {}.  Retry request", mUrl.toString());
+              response = client.execute(request);
+              responseCode = response.getStatusLine().getStatusCode();
+            }
           }
         } catch (IOException e) {
           LOG.error(e.getMessage());
```

Comment:
What happens in this case? We just print it to log and that's it? Try to proceed as usual? I assume that SwiftOutputStream need to throw exception to upper layer, so operations will stop. How to catch that exception when it thrown from the thread?